proto/tendermint/types/ipaddress.go

```diff
@@ -0,0 +1,166 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"inet.af/netaddr"
+)
+
+type IPAddress struct {
+	data netaddr.IP
+}
+
+// Returns underlying netaddr.IP
+func (ip IPAddress) ToNetaddrIP() netaddr.IP {
+	return ip.data
+}
+
+// Parse parses provided IP and sets its value to itself. Supports IPv4 and IPv6.
+func (ip *IPAddress) Parse(address string) error {
```

Comment:
What do you think about changing the design of this component on the following:
Make `Parse`, `MustParse`, and `ParseStdIP` as the functions, rather than the part of the structure IPAddress. It would be better to keep functions like a factory of `IPAddress`. Also, I would recommend returning a value instead of a pointer